### PR TITLE
use a single query to get number of offers completed by user and all …

### PR DESCRIPTION
**Pull request by doodyparizada**
…offer contents together

### CODE REVIEW

File: scripts/src/public/services/offers.ts [hunk 1]
```diff
@@ -47,21 +48,30 @@ function offerDbToApi(offer: db.Offer, content: db.OfferContent) {
...
+		return [];
+	}
+	const offerCounts = await Order.countAllByOffer(userId);
+	const contents = await offerContents.getAllContents();
+
 	return (await Promise.all(
 		app.offers
 			.map(async offer => {
-				if (await offer.didExceedCap(userId)) {
+				if ((offerCounts.get(offer.id) || 0) >= await offer.cap.per_user) {
```
Comment (nitzantomer) [inline scripts/src/public/services/offers.ts]: `await offer.cap.per_user`?
it's not async

File: scripts/src/models/orders.ts [hunk 2]
```diff
@@ -87,6 +87,20 @@ export class Order extends CreationDateModel {
 		return query.getCount();
 	}
 
+	public static async countAllByOffer(userId: string): Promise<Map<string, number>> {
+		const results: Array<{ offer_id: string, cnt: number }> = await getManager().query(
```
[the remark this answers is not in the document]
Comment (nitzantomer) [reply scripts/src/models/orders.ts]: also, why `offer_id` and not `offerId`?

**Comment by nitzantomer** [comment]
did not mean to approve, i meant to ask for changes

File: scripts/src/models/orders.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (doodyparizada) [reply scripts/src/models/orders.ts]: Our convention for the column names are underscores. So it means the variable outside should match the column name. I'll see if it works if I change only the variable name.

File: scripts/src/public/services/offer_contents.ts [hunk 3]
```diff
@@ -99,6 +99,15 @@ export async function getOfferContent(offerId: string, logger: LoggerInstance):
 	return await db.OfferContent.findOne({ offerId });
 }
 
+export async function getAllContents(): Promise<Map<string, db.OfferContent>> {
+	const results = await db.OfferContent.find();
```
[the remark this answers is not in the document]
Comment (doodyparizada) [reply scripts/src/public/services/offer_contents.ts]: sexy

File: scripts/src/public/services/offers.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/public/services/offers.ts]: you're right. Weird that it didn't complain

File: scripts/src/public/services/offer_contents.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (doodyparizada) [reply scripts/src/public/services/offer_contents.ts]: doesn't work

**Pull request merged**